dwave/optimization/model.pyx

```diff
@@ -1358,7 +1358,10 @@ cdef class Symbol:
             inc(it)
 
     def maybe_equals(self, other):
-        """Compare to another node.
+        """Compare to another node. This method exists because a complete equality test can be expensive.
+        A return value of ``0`` guarantees the two nodes are not equal.
```

Comment:
I think we want this under `Returns:`.

I see there are two docstring https://github.com/dwavesystems/dwave-optimization/blob/c36f47b8434251dd12c390827c811d788f9f00b8/dwave/optimization/model.pyx#L1360-L1372 and https://github.com/dwavesystems/dwave-optimization/blob/c36f47b8434251dd12c390827c811d788f9f00b8/dwave/optimization/model.pyx#L1661-L1683

I _think_ if we just delete the docstring for `ArraySymbol.maybe_equals()` it will be inherited from `Symbol.maybe_equals()`.

And we should then copy over
```
        Returns: integer
            Supported return values are:
            
            *   ``0``---Not equal.
            *   ``1``---Might be equal.
            *   ``2``---Are equal.         
```
keeping the best of both worlds.